python/pyarrow/table.pxi

```diff
@@ -1481,6 +1481,8 @@ cdef class Table(_PandasConvertible):
 
         return table
 
+Table = _Table
```

Comment:
I don't understand why you're doing this. The `__name__` of the class will be different, pickle compatibility may be broken... I'd rather keep the original name (i.e. `Table`).